viewer/src/datamodels/cad/CadModelUpdateHandler.ts

```diff
@@ -61,66 +66,13 @@ export class CadModelUpdateHandler {
       this._cameraSubject.pipe(auditTime(250), emissionLastMillis(600)),
       this.loadingModelObservable()
     ]).pipe(
-      observeOn(asyncScheduler),
-      auditTime(250),
-      map(([camera, clippingPlanes, clipIntersection, loadingHints, cameraInMotion, cadModelsMetadata]) => {
-        return {
-          camera,
-          clippingPlanes,
-          clipIntersection,
-          loadingHints,
-          cameraInMotion,
-          cadModelsMetadata
-        };
-      }),
-      filter<DetermineSectorsInput>(
-        ({ cadModelsMetadata, loadingHints }) => cadModelsMetadata.length > 0 && loadingHints.suspendLoading !== true
-      ),
-      publish(input$ => {
-        const modelSectorStates: { [blobUrl: string]: { [id: number]: LevelOfDetail } } = {};
-        const stateHasChanged = filter<WantedSector>(wantedSector => {
-          const sectorStates = modelSectorStates[wantedSector.blobUrl];
-          if (sectorStates) {
-            const sectorState = sectorStates[wantedSector.metadata.id];
-            if (sectorState) {
-              return sectorState !== wantedSector.levelOfDetail;
-            } else {
-              return wantedSector.levelOfDetail != LevelOfDetail.Discarded;
-            }
-          }
-          return true;
-        });
-        const updateSectorState = tap<ConsumedSector>(consumedSector => {
-          let sectorStates = modelSectorStates[consumedSector.blobUrl];
-          if (!sectorStates) {
-            sectorStates = {};
-            modelSectorStates[consumedSector.blobUrl] = sectorStates;
-          }
-          if (consumedSector.levelOfDetail === LevelOfDetail.Discarded) {
-            delete sectorStates[consumedSector.metadata.id];
-          } else {
-            sectorStates[consumedSector.metadata.id] = consumedSector.levelOfDetail;
-          }
-        });
-
-        return input$.pipe(
-          switchMap(input => {
-            const { cameraInMotion } = input;
-            if (cameraInMotion) {
-              return empty();
-            }
-            const wantedSector$ = scheduled(from(sectorCuller.determineSectors(input)), queueScheduler);
-            return wantedSector$.pipe(
-              stateHasChanged,
-              this._sectorRepository.loadSector(),
-              distinctUntilLevelOfDetailChanged(),
-              updateSectorState
-            );
-          })
-        );
-      }),
+      observeOn(asyncScheduler), // Schedule tasks on macro task queue (setInterval)
+      auditTime(250), // Take the last value every 250ms // TODO 07-08-2020 j-bjorne: look into throttle
```

Comment:
Can you elaborate on why we should look into throttle?

Reply:
![image](https://user-images.githubusercontent.com/60690836/89765711-2756d200-daf7-11ea-9dc4-2f6b33fd0809.png)

Throttle with trailing parameter includes the latest emission of the camera position. I am not sure how much difference it would make, but it was just something I came over.

Reply:
Also with the way we have it now with not updating until the camera stops, we could have gone for debounce (but I am expecting we keep it the way it is as we eventually want to change it from not loading, to loading less)